### DISCUSSION TITLE
Some issues regarding computility

### DISCUSSION THREAD

**Discussion by Henry-Lee24** [cyberbotics/wrestling#82] [Q&A] [answered]
I don't know if it's due to computility or other reasons, but some of the actions I can complete on my own computer simulation platform cannot be completed on the competition platform. In addition, I found that on the competition platform, the same action is performed faster when the opponent falls. What is the reason for this.

**Comment by omichel** [comment, marked as answer]
This is probably due to the fact that simulations are asynchronous, e.g., the Webots process doesn't wait until the controller process finishes its control step to proceed with incrementing the simulation time. Actually the Webots process tries to be in sync with real-time only. So, depending on the horsepower and load of the machine, it could be that sometimes, the controller code can complete its step or not. You can get information about it by retrieving the output of the [Robot.step()](https://cyberbotics.com/doc/reference/robot?tab-language=python#wb_robot_step) method. If the return value is 0, it means your control step completed on time successfully, otherwise, there was some delay in the application of your motor commands and possibly the reading of sensor data. In order to benefit from more CPU resources, you may want to use the [Robot.stepBegin() / Robot.stepEnd()](https://cyberbotics.com/doc/reference/robot?tab-language=python#wb_robot_step) methods, so that your controller doesn't wait passively for the answer of Webots, but uses this time to perform some computation in between these calls.

**Reply by Henry-Lee24** [reply]
OK! Thanks.
